60_networks.Rmd

```diff
@@ -0,0 +1,855 @@
+gm["Av.path", "SPRING"] <- average.path.length(spring_graph)
+```
+
+```{r}
+gm
+```
+
+### Network analysis with NetCoMi
+
+The `microNet` object created in section \@ref(netconstruct-netcomi) is now passed to `netAnalyze` to perform network analysis with NetCoMi.
+
+The function computes several common network characteristics such as centrality measures, cluster assignment, the graphlet correlation matrix, as well as global network measures.
+
+The user has several options to choose from, such as a clustering method, how to define hubs, and whether or not to normalize centrality values. See the help page `?netAnalyze` for a description of the arguments.
+
+By default, a heatmap of the Graphlet Correlation Matrix (GCM) is returned (with graphlet correlations in the upper triangle and significance codes resulting from Student's t-test in the lower triangle). See `?calcGCM` and `?testGCM` for details.
+
+```{r netAnalyze_single, fig.width=7, fig.height=7, out.width='75%'}
+spring_netprops <- netAnalyze(spring_net, 
+                              clustMethod = "cluster_fast_greedy",
+                              hubPar = "eigenvector",
+                              normDeg = FALSE)
+```
+
+```{r summary_single}
+summary(spring_netprops, numbNodes = 5)
+```
+
+**Interpretation of some findings:**
+
+-   The largest connected component (LCC) has 103 nodes and the network contains 42 singletons.
+-   9 clusters have been identified, containing 2 to 16 nodes.
+-   There are eight hub nodes detected, which by definition are the nodes with the highest eigenvector centrality.
+-   The average path length is 3.81. This means that on average it takes 3.81 steps (step length is the average dissimilarity) to get from one node to another node in the network. Note that the average path length in NetCoMi is defined differently than in the igraph package, which is why the values differ.
+-   Low values of edge density and the connectivity measures indicate that the network is rather sparse and not robust to perturbations (i.e., removal of nodes or edges).
+
+## Network visualization
+
+Further insight into the network structure can be gained by visualizing the network, which is done using NetCoMi's plot function. The help page can be accessed via `?plot.microNetProps`.
+
+### Highlight node properties
+
+In the first plot, node colors represent the detected clusters and node sizes are scaled by eigenvector centrality. Hub nodes are highlighted by default. Singletons are not included in the plot. To improve the readability, NetCoMi's "intelligent" label shortening approach is used.
+
+Note that nodes are sometimes placed too close together so that the labels overlap. You may need to play around with the repulsion argument until you find a value where the labels are legible, but also the clusters are still well recognizable.
+
+```{r network_plot_single_cluster, fig.width=13, fig.height=10}
+plot(spring_netprops,
```

Comment:
No ggplot equivalent available?

Reply:
NetCoMi comes with it's own plot function, which is based on qgraph, which is another state-of-the-art package for network plotting. Considerable effort has been put into ensuring that the function can be used to create visually appealing network plots.

Reply:
The idea is also that the plot function expects the object returned by netAnalyze() as input. Thus, it contains all the results of the network analysis, and the user can simply change some arguments in the plot function to highlight different network measures.